aria/modeling/service_instance.py

```diff
@@ -529,6 +524,32 @@ def validate_capabilities(self):
                 satisfied = False
         return satisfied
 
+    def find_host(self):
+        def _find_host(node):
```

Comment:
If we do that then we'll be calling `find_host` again on again on the same nodes that are targets of different relationships. So, to do that efficiently we would need some extra flag (`found_host`?) to signify that we don't have to do that again for that node. It made more sense to me to have it set once and only once per node.